### PR TITLE
Use new scheduler and handle future specs

### PR DESCRIPTION
**Pull request by we-ai**
This PR improves [PR#415](https://github.com/episphere/connectFaas/pull/415)(for issue [#695](https://github.com/episphere/connect/issues/695)) and [PR#401](https://github.com/episphere/connectFaas/pull/401)(for issues [#692](https://github.com/episphere/connect/issues/692)). Below are details:
- Use a new scheduler in each tier for notifications backups
- Handle future new notification specifications when doing BQ queries

### CODE REVIEW

File: utils/bigquery.js [hunk 1]
```diff
@@ -62,23 +62,36 @@ async function getParticipantsForNotificationsBQ({
...
+    bqFieldArray.length === 0 ? "*" : bqFieldArray.join(", ")
+  } FROM \`Connect.participants\` WHERE ${bqConditionArray.join(" AND ")}`;
 
   try {
+    const queryStr = `${queryStrCommon} AND query.notificationSpecIdsUsed.${bqNotificationSpecId} IS NOT true LIMIT ${limit} OFFSET ${offset}`;
     const [rows] = await bigquery.query(queryStr);
     if (rows.length === 0) return result;
 
     result.hasNext = rows.length === limit;
     result.fetchedDataArray = rows.map((data) => convertToFirestoreData(data));
```
Comment (JoeArmani) [inline utils/bigquery.js:75]: Maybe this can be turned into a function to avoid code duplication? 71-75, 81-85

**Review by JoeArmani — APPROVED** [review]
looks good. I learned a bit about BQ going through this (haven't worked with it yet). Thanks!

**Review by anthonypetersen — APPROVED** [review]
looks good

File: utils/bigquery.js (hunk 1 above)
Comment (we-ai) [reply utils/bigquery.js:75]: I thought about it. 
Current the 4-line code has 2 copies (total 8 lines without empty lines). 
If I add a new function to remove 4 lines of duplicate, the newly created function will be 8 lines, plus 2 lines of function calls. There'll be 10 lines of code after creating the function.
So, I'd defer the new function to a later time when the 4-line code is used more frequently.

**Pull request merged**